app/assets/javascripts/reducers/users.js

```diff
@@ -6,6 +6,8 @@ const initialState = {
   sort: {
     sortKey: null,
     key: null,
+    initialSortKey: 'real_name',
```

Comment:
Uhm, I think I can just set the default value in the reducer instead.

Reply:
I don't think that helps solve the problem here, which is that user-selected sorting gets overridden when RECEIVE_USERS happens (again, after the initial action followed by a sorting action that the user triggered). you can just set the initial value of  `sort.sortKey`, and then use that (which may have changed based on a subsequent SORT_USERS action) every time within the RECEIVE_USERS reducer.

Reply:
Would it be better if I did the default sorting in the SORT_USERS instead? To avoid the overriding by the RECEIVE_USERS?

Reply:
I don't think I understand what you're suggesting. Your current implementation already does default sorting in RECEIVE_USERS.

Reply:
Yes. I was wondering whether I should move the default sorting to SORT_USERS instead to avoid the overidding. But also isn't RECEIVE_USERS only called on refresh? Which then I think should reset the sorting to default? 
I think I'm a bit confused, I hope my communication is clear.

Reply:
![Screenshot from 2023-04-13 13-41-04](https://user-images.githubusercontent.com/88780311/231840873-d2cb5670-8f59-4389-bff2-c7d17659e453.png)
Something like this?

Reply:
There are two situations where RECEIVE_USERS action happens:
1. when a course page is rendered initially, before the client has received any data about the users in the course. No sorting initiated by the user can take place before the initial user data is received. 
2. when at least 1 minute has passed and the user navigates between tabs, and the client automatically requests a refresh of user data. this is the case where we don't want the sorting to get reset. there might be additional users who were not part of the course previously, and so this might result in changes to the user list... but it should not change the sorting, including in the case where the user selected their preferred non-default sorting.

Reply:
Okay, this is clearer now. Let me update it. Thank you.